pkg/util/duration/duration.go

```diff
@@ -621,7 +621,7 @@ func (d Duration) DivFloat(x float64) Duration {
 	dayInt, dayFrac := math.Modf((float64(d.Days) / x) + (monthFrac * DaysPerMonth))
 
 	return MakeDuration(
-		int64((float64(d.nanos)/x)+(dayFrac*float64(nanosInDay))),
+		int64(float64((time.Duration(float64(d.nanos)/x) * time.Nanosecond).Truncate(time.Microsecond).Nanoseconds())+(dayFrac*float64(nanosInDay))),
```

Comment:
Can you pull out `(time.Duration(float64(d.nanos)/x) * time.Nanosecond).Truncate(time.Microsecond).Nanoseconds()` into a separate variable to make it easier to read, i.e.:

```
// We ensure we round down nanoseconds when doing division to
// match PostgreSQL.
retNanos := time.Duration(float64(d.nanos)/x).Truncate(time.Milliseconds)

return MakeDuration(
   ...,
  float64(retNanos.Nanoseconds()) + (dayFrac*float64(nanosInDay),
   ...,
)
```